bash_scripting_102/bash_scripting_102.md

```diff
@@ -0,0 +1,477 @@
+
+```
+mkdir color_name_animals
+```
+
+If we want to copy the entire directory of `blue_animals` into this new folder, we need to use the `-r` flag:
+
+```
+cp -r blue_animals color_name_animals
+```
+
+<div class = "important">
+Moving a directory with `mv` will move all of its contents to the new location.
+
+Copying a directory with `cp` requires the `-r` flag in order to copy all of its contents.
+</div>
+
+### Quiz: Organizing files
+
+1. You create a new folder titled `canines` in `learning_bash-main`. What command would you use to move the files `dog.txt` and `wolf.txt` into the `canines` folder? Use the relative locations of the files and folders.
+
+[[mv dog.txt wolf.txt canines]]
+<script>
+  let input = "@input";
+  input == "mv dog.txt wolf.txt canines" || input == "cmv wolf.txt dog.txt canines";
+</script>
+***
+<div class = "answer">
+The command `mv dog.txt wolf.txt canines` will move both files into the `canines` folder. The order of `dog.txt` and `wolf.txt` doesn't matter, but `mv` must be first and `canines`, the target folder, must be last.
+</div>
+***
+
+2. Which of these commands would you use to put a copy of the file `coyote.txt` into the `canines` folder?
+
+[[ ]] `mv coyote.txt canines`
+[[X]] `cp coyote.txt canines`
+[[ ]] `mv -r coyote.txt canines`
+[[ ]] `cp -r coyote.txt canines`
+***
+<div class = "answer">
+The command `mv coyote.txt canines` will move the entire file into the `canines` folder.
+
+The command `cp coyote.txt canines` will create a copy of `coyote.txt` in the folder `canines`.
+
+The command `mv -r coyote.txt canines` will raise an error since `-r` is not one of the options of the command `mv`.
+
+The command `cp -r coyote.txt canines` will **also** create a copy of `coyote.txt` in the folder `canines`. However the flag `-r` is not necessary here because `coyote.txt` is a file and not a directory.
+</div>
+***
+
+## Deleting files
+
+If you are done with a file and sure you won't need it ever again, it might make sense to delete it.
+
+The command `rm` **remove**s and files listed after it from your computer. Maybe you don't actually want two different names for the same species of bear and want to delete the `brown_bear.txt` file we made earlier:
+
+```
+rm brown_bear.txt
+```
+
+<div class = "warning">
+Removing files from your command line interface with the `rm` command **removes** them from your computer. It does not move them to your "Trash" folder.
+
+Be completely sure that you want to remove a file before using the `rm` command.
+</div>
+
+If you are sure you want to delete a directory, the recursive flag `-r` will recursively delete a folder and all of its contents, and the contents of any subfolders, and on an on until everything is gone.
+
+Maybe you no longer want a `red_animals` folder since all of its contents are available elsewhere:
+
+```
+rm -r red_animals
+```
+
+### Quiz: Deleting files
+
+Which statements about the remove command `rm` are true?
```

Comment:
I'd recommend making the "true" stand out here, either by bolding it or making it all-caps.

Reply:
BOTH!